reporter/http/http.go

```diff
@@ -150,7 +157,10 @@ func (r *httpReporter) sendBatch() error {
 		r.reqCallback(req)
 	}
 
-	resp, err := r.client.Do(req)
+	timeoutCtx, cancelFn := context.WithTimeout(req.Context(), r.reqTimeout)
```

Comment:
nit:
s/timeoutCtx/ctx
s/cancelFn/cancel

as they seem to be more idiomatic names for those

Reply:
agree here with the suggestion.. nothing it collides with.